zummon/src/commonMain/kotlin/companysurvey/Validation.kt

```diff
@@ -16,231 +16,288 @@ enum class Status {
         }
     }
-    ValidationResult(Status.NOT_APPLICABLE, "Small consumption validations are not applicable")
-}
 
-// Validator for power per charge point in range 3..150 kW
-val validatePowerPerChargeCars = Validator { survey: Survey ->
-    val gridConnection = survey.getSingleGridConnection()
-    val powerPerChargePointCars = gridConnection.transport.cars.powerPerChargePointKw
+    // Validator for contracted feed-in capacity <= physical capacity
+    fun validateContractedFeedInCapacity(electricity: Electricity): ValidationResult {
+        val feedInCapacity = electricity.getContractedFeedInCapacityKw()
+        val physicalCapacity = electricity.getPhysicalConnectionCapacityKw()
 
-    when {
-        powerPerChargePointCars == null -> ValidationResult(Status.NOT_APPLICABLE, "Cars Power per charge point is not provided")
-        powerPerChargePointCars in 3.0..150.0 -> ValidationResult(Status.VALID, "Cars Power per charge point is valid")
-        else -> ValidationResult(Status.INVALID, "Cars power per charge point ${powerPerChargePointCars} is outside the valid range (3..150 kW)")
+        return when {
+            feedInCapacity == null -> ValidationResult(Status.MISSING_DATA, "No feed-in capacity data")
+            physicalCapacity == null -> ValidationResult(Status.MISSING_DATA, "No physical capacity data")
+            feedInCapacity <= physicalCapacity -> ValidationResult(Status.VALID, "Feed-in capacity is valid")
+            else -> ValidationResult(Status.INVALID, "Feed-in capacity ${feedInCapacity} exceeds physical capacity ${physicalCapacity}")
+        }
     }
-}
 
-val validatePowerPerChargeTrucks = Validator { survey: Survey ->
-    val gridConnection = survey.getSingleGridConnection()
-    val powerPerChargePointTrucks = gridConnection.transport.trucks.powerPerChargePointKw
+    // Validator for PV production >= feed-in
+    fun validatePvProductionFeedIn(electricity: Electricity): ValidationResult {
+        val annualProduction = electricity.annualElectricityProduction_kWh
+        val feedIn = electricity.annualElectricityFeedIn_kWh
 
-    when {
-        powerPerChargePointTrucks == null -> ValidationResult(Status.NOT_APPLICABLE, "Trucks Power per charge point is not provided")
-        powerPerChargePointTrucks in 3.0..150.0 -> ValidationResult(Status.VALID, "Trucks Power per charge point is valid")
-        else -> ValidationResult(Status.INVALID, "Trucks power per charge point ${powerPerChargePointTrucks} is outside the valid range (3..150 kW)")
+        return when {
+            annualProduction == null -> ValidationResult(Status.MISSING_DATA, "No PV production data")
+            feedIn == null -> ValidationResult(Status.MISSING_DATA, "No feed-in data")
+            annualProduction >= feedIn -> ValidationResult(Status.VALID, "PV production is valid")
```

Comment:
condition should be flipped

Reply:
Annual pv production should be more than annual feed-in, isn't it?

Reply:
Yes sorry